### DISCUSSION TITLE
Download images older than x days

### DISCUSSION THREAD

**Discussion by roberto-formulatrix** [icloud-photos-downloader/icloud_photos_downloader#820] [Ideas]
I thought it would be cool to be able to download (and delete) images that are older than xxx days on the icloud side.  The idea here is that the older the pictures are they're probably not being seen much on your phone so why not download the pictures out of the icloud and have them available for "offline" view?

If I were to state the requirements, it would be like this:
   - add parameter maxdays (int: # days).  Pictures newer than that will not get downloaded / deleted /affected
   - add parameter mincount (int: # pics). # of pictures to leave online.  E.g. if you set it to 1000 and only have 2000 images (and they're all older olderthan maxdays), then the latest 1000 are not downloaded/affected, but the rest are since it they match all criteria.
   -  Need a way to keep picture online... maybe if they're hearted, or are in a special folder, or are shared?

**Comment by roberto-formulatrix** [comment]
My ideal scenario: run it every night, download/delete anything older than 90 days but keep the latest 1000 on the cloud..

**Reply by porjo** [reply]
Duplicate of #659 ?

**Comment by AndreyNikiforov** [comment]
> My ideal scenario: run it every night, download/delete anything older than 90 days but keep the latest 1000 on the cloud..

IIUC you want photos local collection always be 90 days behind online one. I am curious what is the reason to NOT want latest online photos in your local collection.

**Reply by roberto-formulatrix** [reply]
hmm, the idea is that likely the ones that I took recently (or the ones I love) are the ones that I'd like to have with easy access from my phone.  I think of the local storage more like my archive.  This way I also ensure that my icloud space has space freely available to it.

Having said that, is there a way to delete the pictures after download?  I think I'm noticing that the pics don't get deleted   --auto-delete   means deletion from the "Recently deleted" folder (so more of a cleanup; nothing to do with the photos downloaded)

Is there a way to do what I intend already?

**Reply by AndreyNikiforov** [reply]
There is a way to delete photos in icloud after they have been downloaded to your local storage with `--delete-after-download` flag (==move from icloud to local). However,  it does not take additional date constraints, so anything that is downloaded will be deleted in icloud.

I assume, ideally, you want photos older than X days to be moved from icloud to local, while photos newer than X days copied/synchronized from icloud to local. There is no way to specify time constraint on downloaded photos as older than X days, so it is not possible to achieve your goal IIUC.

**Reply by roberto-formulatrix** [reply]
Ah, I missed that flag - thanks!

Could it be made so that it allows us to give a date or maybe simply a number?  E.g. older than x days?. 
Or maybe even better, only delete files downloaded if older than x days. (i.e. our local disk is both an archive but has the latest... so I can start to reorganize them).
 Sounds like a nifty feature :)

**Comment by stanthewizzard** [comment]
is there a way to download from dateA to dateB ?
